app/controllers/users/sessions_controller.rb

```diff
@@ -1,2 +1,13 @@
 class Users::SessionsController < Devise::SessionsController
+  def new
+    super do |resource|
+      # When warden and devise threw an authentication error (invalid password)
+      # They don't respect the locale param
+      # This allows us to detect, enforce internationalization, and move the error message
+      if flash[:alert] == I18n.t("devise.failure.invalid", locale: :en) # The default english failure message
+        flash.delete :alert
+      end
+      resource.errors[:password] << I18n.t("devise.failure.invalid")
```

Comment:
This should be inside the conditional

Reply:
It'd be easier for me to understand if we use a different translation key for our message. Then conceptually,  we're looking for their string's presence & replacing it with our entirely different thing.